### DISCUSSION TITLE
Is it possible to have a Win11-style context menu?

### DISCUSSION THREAD

**Discussion by myfix16** [HavenDV/H.NotifyIcon#84] [Ideas]
Currently, for the context menu, we have either a Win32 PopupMenu or a custom one (`ContextMenuMode="SecondWindow"`). Is it possible to add support for Win11-style context menus (like the context menu of the system volume control)?
<img width="128" alt="image" src="https://user-images.githubusercontent.com/61649477/228306284-914b0d2e-0fbc-4760-9d75-4b46d8740e5f.png">

**Comment by HavenDV** [comment]
I think it's possible, but I don't know how. My knowledge of win32 is not enough to customize the popup window so that it looks modern

**Reply by myfix16** [reply]
I think https://github.com/HavenDV/H.NotifyIcon/discussions/79#discussioncomment-5216111 this discussion provides a good aspect to work with. Right-clicking on the icon will display a custom window containing a XAML context menu. At least it should work for WinUI3.

**Reply by brandon3343** [reply]
> I think it's possible, but I don't know how. My knowledge of win32 is not enough to customize the popup window so that it looks modern

we can do that when windowsAppSDK 1.3 is released, At that time we can add acrylic backdrop easier than now. 

when acrylic backdrop is not supported, fallback to default style.

just like that.
![1680647859030](https://user-images.githubusercontent.com/17621099/229937703-ffeb680c-e690-47a8-9a15-54122ebe0e66.jpg)

**Reply by myfix16** [reply]
@l619534951 That's cool. What is the app in the picture?

**Reply by brandon3343** [reply]
"MonitorHelper", it's not release yet, you can use "twinkle Tray" instead.

**Reply by Anthonyy232** [reply]
@brandon3343 Hi Brandon, do you have a GitHub link to that app? I'm interested in look at how they implemented such a thing.

**Reply by brandon3343** [reply]
Sorry, I’m no longer using WinUI, and I’ve deleted the source code.
This feature is implemented via DDC/CI.
On Windows, it can be achieved using the [monitor-configuration](https://learn.microsoft.com/zh-cn/windows/win32/monitor/monitor-configuration) API.
If a certain API is not provided, you can send the VCP CODE manually to implement it.

**Reply by Anthonyy232** [reply]
Thanks for the reply! Sorry I did not clarify, I am interested in how to implement the pop-up like that as it looks very nice. If you have any pointers, that would be awesome. 

I'm curious, why did you stop using WinUI and delete it? 

Thanks.

**Reply by brandon3343** [reply]
It is simply a regular window with the title bar hidden and an acrylic backdrop applied.
